kubernetes-itests/src/test/java/io/fabric8/commons/Delete.java

```diff
@@ -0,0 +1,57 @@
+package io.fabric8.commons;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+import java.util.concurrent.Callable;
+
+public class Delete<T> implements Callable<Boolean> {
+
+  private KubernetesClient client;
+  private String name;
+  private String namespace;
+  private Class<T> typeParameterClass;
+
+  public Delete(Class<T> typeParameterClass, KubernetesClient client, String name, String namespace) {
+    this.typeParameterClass = typeParameterClass;
+    this.client = client;
+    this.name = name;
+    this.namespace = namespace;
+  }
+
+  @Override
+  public Boolean call() throws Exception {
+    switch (this.typeParameterClass.getSimpleName()) {
+      case "Route":
+        return ((OpenShiftClient)this.client).routes().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "Template":
+        return ((OpenShiftClient)this.client).templates().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "ImageStream":
+        return ((OpenShiftClient)this.client).imageStreams().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "DeploymentConfig":
+        return ((OpenShiftClient)this.client).deploymentConfigs().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "BuildConfig":
+        return ((OpenShiftClient)this.client).buildConfigs().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "ConfigMap":
+        return this.client.configMaps().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "StatefulSet":
+        return this.client.apps().statefulSets().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "Deployment":
+      return this.client.apps().deployments().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "Service":
+        return this.client.services().inNamespace(this.namespace).list().getItems().size() == 0;
+      case "ServiceAccount":
```

Comment:
Why is service account name hardcoded here?

Reply:
oops